src/main/java/com/builtbroken/bashrockmakefire/logic/EventHandler.java

```diff
@@ -22,63 +22,52 @@
                             {
-                                //Check if we can place fire
-                                final BlockPos firePos = clickPos.up();
-                                final IBlockState blockAbove = event.getWorld().getBlockState(firePos);
-                                if (blockAbove.getBlock().isReplaceable(event.getWorld(), firePos))
-                                {
-                                    event.setCanceled(true);
-                                    event.getWorld().setBlockState(firePos, Blocks.FIRE.getDefaultState());
-                                    if (event.getEntityPlayer().getGameProfile() != null)
-                                    {
-                                        BashFireMakeRock.LOGGER.info(
-                                                event.getEntityPlayer().getGameProfile().getName() + " [" + event.getEntityPlayer().getGameProfile().getId() + "]"
-                                                        + " created fire at " + firePos
-                                                        + " in world " + event.getWorld().getWorldInfo().getWorldName()
-                                                        + " dim: " + event.getWorld().provider.getDimension());
-                                    }
-                                }
-
-                                //Remove as we are done
-                                clickDataHashMap.remove(clickPos);
+                                BashFireMakeRock.LOGGER.info(
+                                        event.getEntityPlayer().getGameProfile().getName() + " [" + event.getEntityPlayer().getGameProfile().getId() + "]"
+                                                + " created fire at " + firePos
+                                                + " in world " + event.getWorld().getWorldInfo().getWorldName()
+                                                + " dim: " + event.getWorld().provider.getDimension());
                             }
                         }
-                        else
-                        {
-                            clickDataHashMap.put(clickPos, new ClickData(clickPos));
-                        }
+
+                        //Remove as we are done
+                        clickDataHashMap.remove(clickPos);
                     }
                 }
-
-                //Clear dead entries TODO check for performance hit on larger servers
-                clickDataHashMap.keySet().stream()
-                        .filter(e -> clickDataHashMap.get(e).isDead())
-                        .collect(Collectors.toList())
-                        .forEach(e -> clickDataHashMap.remove(e));
+                else
+                {
+                    clickDataHashMap.put(clickPos, new ClickData(clickPos));
+                }
             }
+
+            //Clear dead entries
+            clickDataHashMap.entrySet().removeIf(e->e.getValue().isDead());
```

Comment:
Good change, that previous code was a mess

Reply:
Oof, thanks ! I wasn't aware that's a possibility ! 👍

Reply:
Ya i've overlooked it a few times myself.